redfish-core/include/utils/hw_isolation.hpp

```diff
@@ -0,0 +1,337 @@
+ *         HardwareIsolation manager may be used the "Resolved" dbus entry
+ *         property to indicate the deisolation instead of delete
+ *         the entry object.
+ */
+inline void
+    deisolateResource(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                      const sdbusplus::message::object_path& resourceObjPath,
+                      const std::string& hwIsolationDbusName)
+{
+    // Get the HardwareIsolation entry by using the given resource
+    // associations endpoints
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, resourceObjPath, hwIsolationDbusName](
+            boost::system::error_code& ec,
+            const std::variant<std::vector<std::string>>& vEndpoints) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR(
+                "DBus response error [{} : {}] when tried to get the hardware isolation entry for the given resource dbus object path: ",
+                ec.value(), ec.message(), resourceObjPath.str);
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        std::string resourceIsolatedHwEntry;
+        const std::vector<std::string>* endpoints =
+            std::get_if<std::vector<std::string>>(&(vEndpoints));
+        if (endpoints == nullptr)
+        {
+            BMCWEB_LOG_ERROR(
+                "Failed to get Associations endpoints for the given object path: {}",
+                resourceObjPath.str);
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        resourceIsolatedHwEntry = endpoints->back();
+
+        // De-isolate the given resource
+        crow::connections::systemBus->async_method_call(
+            [asyncResp,
+             resourceIsolatedHwEntry](const boost::system::error_code& ec1,
+                                      const sdbusplus::message::message& msg) {
+            if (!ec1)
+            {
+                messages::success(asyncResp->res);
+                return;
+            }
+
+            BMCWEB_LOG_ERROR(
+                "DBUS response error [{} : {}] when tried to isolate the given resource: {}",
+                ec1.value(), ec1.message(), resourceIsolatedHwEntry);
+
+            const sd_bus_error* dbusError = msg.get_error();
+
+            if (dbusError == nullptr)
+            {
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            BMCWEB_LOG_ERROR("DBus ErrorName: {} ErrorMsg: {}", dbusError->name,
+                             dbusError->message);
+
+            if (strcmp(dbusError->name,
+                       "xyz.openbmc_project.Common.Error.NotAllowed") == 0)
+            {
+                messages::propertyNotWritable(asyncResp->res, "Entry");
+            }
+            else if (strcmp(dbusError->name,
+                            "xyz.openbmc_project.Common.Error.Unavailable") ==
+                     0)
```

Comment:
The use of `strcmp()`  can also be replaced using `std::string_view()` comparison like L61 -- probably in the next Jira item

```
            if (std::string_view("xyz.openbmc_project.Common.Error.NotAllowed") == dbusError->name)
```